src/main/java/chess/domain/ChessGame.java

```diff
@@ -0,0 +1,61 @@
+package chess.domain;
+
+import chess.db.entity.ChessGameEntity;
+import chess.domain.piece.Color;
+import chess.domain.piece.Piece;
+import chess.domain.piece.Position;
+import java.util.List;
+import java.util.Map;
+
+public class ChessGame {
+
+    private final ChessBoard chessBoard;
+    private Color currentColor;
+
+    public ChessGame() {
+        this.chessBoard = ChessBoardMaker.init();
+        this.currentColor = Color.WHITE;
+    }
+
+    public ChessGame(final ChessBoard chessBoard) {
+        this.chessBoard = chessBoard;
+        this.currentColor = Color.WHITE;
+    }
+
+    public ChessGame(final ChessGameEntity chessGameEntity) {
+        this.chessBoard = ChessBoard.fromEntity(chessGameEntity.chessBoard());
+        this.currentColor = chessGameEntity.currentColor();
+    }
```

Comment:
이렇게 된다면 도메인 객체가 db 레이어 객체에 의존성을 가지게 되지 않을까요?

Reply:
반영하여 변환을 DTO 내에서 해주도록 변경해보았습니다.
```suggestion
    public ChessGame(final ChessGameDto chessGameDto) {
        this.chessBoard = chessGameDto.getChessBoard();
        this.currentColor = chessGameDto.getCurrentColor();
    }
```